sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -265,8 +265,13 @@ object FileFormat {
    * fields of the [[PartitionedFile]], and do have entries in the file's metadata map.
    */
   val BASE_METADATA_EXTRACTORS: Map[String, PartitionedFile => Any] = Map(
-    FILE_PATH -> { pf: PartitionedFile => pf.toPath.toString },
-    FILE_NAME -> { pf: PartitionedFile => pf.toPath.getName },
+    FILE_PATH -> { pf: PartitionedFile =>
+      // Use new Path(Path.toString).toString as a form of canonicalization
+      new Path(pf.filePath.toPath.toString).toString
```

Comment:
oh wait, does it return encoded path string? The test only checks file_name but not file_path

Reply:
I think we should generate the path string from a URI if we want it to be encoded.

Reply:
Whoops, this is what I get for doing all my development in the middle of the night 🤦 
Sorry, I will put up a new PR soon.

Reply:
It's OK, 3.4.1 won't be released soon and we have time for a followup PR. In the followup PR let's also test `file_path` as well.

Reply:
https://github.com/apache/spark/pull/41054